--- a/drivers/kernelsu/core_hook.c
+++ b/drivers/kernelsu/core_hook.c
@@ -1098,7 +1098,7 @@
 #endif
 }
 
-static void try_umount(const char *mnt, bool check_mnt, int flags)
+static void try_umount(const char *mnt, bool check_mnt, int flags, uid_t uid)
 {
 	struct path path;
 	int err = kern_path(mnt, 0, &path);
@@ -1285,25 +1285,25 @@
 
 	// fixme: use `collect_mounts` and `iterate_mount` to iterate all mountpoint and
 	// filter the mountpoint whose target is `/data/adb`
-	try_umount("/odm", true, 0);
-	try_umount("/system", true, 0);
-	try_umount("/system_ext", true, 0);
-	try_umount("/vendor", true, 0);
-	try_umount("/product", true, 0);
-	try_umount("/data/adb/modules", false, MNT_DETACH);
+	try_umount("/odm", true, 0, new_uid.val);
+	try_umount("/system", true, 0, new_uid.val);
+	try_umount("/system_ext", true, 0, new_uid.val);
+	try_umount("/vendor", true, 0, new_uid.val);
+	try_umount("/product", true, 0, new_uid.val);
+	try_umount("/data/adb/modules", false, MNT_DETACH, new_uid.val);
 
 	// try umount ksu temp path
-	try_umount("/debug_ramdisk", false, MNT_DETACH);
-	try_umount("/sbin", false, MNT_DETACH);
+	try_umount("/debug_ramdisk", false, MNT_DETACH, new_uid.val);
+	try_umount("/sbin", false, MNT_DETACH, new_uid.val);
 
 	// try umount hosts file
-	try_umount("/system/etc/hosts", false, MNT_DETACH);
+	try_umount("/system/etc/hosts", false, MNT_DETACH, new_uid.val);
 
 	// try umount lsposed dex2oat bins
-	try_umount("/apex/com.android.art/bin/dex2oat64", false, MNT_DETACH);
-	try_umount("/apex/com.android.art/bin/dex2oat32", false, MNT_DETACH);
+	try_umount("/apex/com.android.art/bin/dex2oat64", false, MNT_DETACH, new_uid.val);
+	try_umount("/apex/com.android.art/bin/dex2oat32", false, MNT_DETACH, new_uid.val);
 
 	return 0;
 }
@@ -1550,5 +1550,3 @@
 	// ksu_kprobe_exit();
 #endif
 }
-
-}
